### PR TITLE
Update the in article tracking subtype to provide more detail

### PR DESCRIPTION
**Pull request by remybach**

### CODE REVIEW

File: server/middleware/display-variants.js [hunk 1]
```diff
@@ -13,6 +14,10 @@ module.exports = (req, res, next) => {
 		type = 'registration'
 	}
 
+	if (req.headers['ft-flags'].indexOf('inArticlePreview') !== -1) {
```
Comment (tavvy) [inline server/middleware/display-variants.js]: We should be able to read the flag via `res.locals.flags` which is the flags-api-client.
Then we could set the subtype with `${type}-${res.locals.flags('inArticlePreview')}`

Comment (remybach) [inline server/middleware/display-variants.js]: Ah, of course. I'll update this. 👍

**Comment by tavvy** [comment]
This maybe ok for now.
I think ideally this information should be returned from the `barrier-guru` and then used to set the value.
It isn't scalable to be using the flag to set the type.
Maybe barrier-guru can return `subType` as a top level property for all the barrier variants?

**Comment by remybach** [comment]
I've updated this to now grab the subtype from what barrier-guru gives us.

Depends on https://github.com/Financial-Times/next-barrier-guru/pull/87

File: views/partials/barriers/shared/grid.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div class="trial-barrier-grid__options o-grid-row o-grid-row--compact trial-barrier-grid__options--{{numberOfOffers}}"
 	data-trackable="trial-barrier-grid"
 	data-barrier="{{display.type}}"
-	data-opportunity-subtype="{{display.type}}"
+	data-opportunity-subtype="{{subtype}}"
```
Comment (tavvy) [inline views/partials/barriers/shared/grid.html]: As this is a shared template we need to have a solution for the non `truncated-article` subTypes.

So replicate how `display.type` was set by`req.params.type ? req.params.type.toLowerCase() : 'trial';` from `display-variant.js`

We should set the subtype with logic along the lines of:
`barrier.subType ? barrier.subType : req.params.type`

Then use it here.

Either that or update all barrier-guru configs to return a subType for use.

Comment (remybach) [inline views/partials/barriers/shared/grid.html]: Updated the barrier-guru configs as discussed.

**Pull request merged**